### PR TITLE
Shopping cart us26

### LINKED ISSUE #30: User Story 26, Registered users can check out
As a registered user
When I add items to my cart
And I visit my cart
I see a button or link indicating that I can check out
And I click the button or link to check out and fill out order info and click create order
An order is created in the system, which has a status of "pending"
That order is associated with my user
I am taken to my orders page ("/profile/orders")
I see a flash message telling me my order was created
I see my new order listed on my profile orders page
My cart is now empty

### PR DESCRIPTION
**Pull request by priyapower**
closes #30 

**Contributors**
- Driver(s): Luke, Priya
- Navigator(s): Luke, Priya, Leah, Grant
  
**What was changed:**
- This updates the shopping cart logic to only show checkout when you are logged in (might be a us25 update)
- This adds user_id foreign key to order database (with all spec and model updates)
- This adds .status? default false for orders database
- This updates ALL SPECS for bangs at end of creates and unique emails when necessary
- This creates user action ```orders``` and new view page ```order.html.erb``` nested under users

**Any tech notes (if applicable)**
- Haven't checked anything in rails server or Heroku deployment to confirm CSS/HTML - made minor updates to html/ruby code and tested with ```save_and_open_page```

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -1,6 +1,8 @@
 class Order <ApplicationRecord
   validates_presence_of :name, :address, :city, :state, :zip
+  validates_inclusion_of :status?, :in => [true, false]
```
Comment (LHJE) [inline app/models/order.rb:3]: I'm glad we were able to get this to work.

File: app/models/user.rb [hunk 2]
```diff
@@ -1,6 +1,10 @@
 class User < ApplicationRecord
+  has_many :orders
 
-  validates_presence_of :name, :address, :city, :state, :zip, :password, :password_confirmation, :role
+  validates_presence_of :name, :address, :city, :state, :zip, :role
+
+  validates :password, :presence => true, :confirmation => true, :on => :create
+  validates :password_confirmation, :presence => true, :confirmation => true, :on => :create
```
Comment (LHJE) [inline app/models/user.rb:7]: Very cool!

File: app/views/users/order.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  <% @orders.each do |order| %>
+  <ul>
+    <%= link_to "Order: #{order.id}", "/orders/#{order.id}" %>
+  </ul>
+  <ul>
+    <% if order.status? %>
+      <p>Status: Shipped</p>
+    <% else %>
+      <p>Status: Pending</p>
+    <% end %>
```
Comment (LHJE) [inline app/views/users/order.html.erb:11]: Look good and dry!

File: config/routes.rb [hunk 4]
```diff
@@ -41,9 +41,14 @@
   post "/orders", to: "orders#create"
   get "/orders/:id", to: "orders#show"
```
Comment (LHJE) [inline config/routes.rb:43]: Let's take out this extra line.

File: spec/features/cart/show_spec.rb [hunk 5]
```diff
@@ -4,12 +4,12 @@
   describe 'When I have added items to my cart' do
     describe 'and visit my cart path' do
       before(:each) do
-        @mike = Merchant.create(name: "Mike's Print Shop", address: '123 Paper Rd.', city: 'Denver', state: 'CO', zip: 80203)
-        @meg = Merchant.create(name: "Meg's Bike Shop", address: '123 Bike Rd.', city: 'Denver', state: 'CO', zip: 80203)
+        @mike = Merchant.create!(name: "Mike's Print Shop", address: '123 Paper Rd.', city: 'Denver', state: 'CO', zip: 80203)
```
Comment (LHJE) [inline spec/features/cart/show_spec.rb:7]: It took a while, but we did get them all, and it was a good idea.

File: spec/features/users/edit_spec.rb [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+      click_link "Change Password"
+
+      fill_in(:password, with: "newpassword")
+      fill_in(:password_confirmation, with: "doesnotmatch")
+      click_button "Change Password"
+
+      expect(current_path).to eq("/profile/edit")
+      expect(page).to have_content("Password confirmation doesn't match Password")
+    end
+  end
```
Comment (LHJE) [inline spec/features/users/edit_spec.rb:82]: A good rigorous test. Nicely done!

File: spec/models/order_spec.rb [hunk 7]
```diff
@@ -7,22 +7,24 @@
     it { should validate_presence_of :city }
     it { should validate_presence_of :state }
     it { should validate_presence_of :zip }
+    it { should validate_inclusion_of(:status?).in_array([true,false]) }
```
Comment (LHJE) [inline spec/models/order_spec.rb:10]: It was great when this finally came together!

**Review by LHJE — APPROVED** [review]

File: app/models/order.rb (hunk 1 above)
Comment (priyapower) [reply app/models/order.rb:3]: Me too! I still have so many questions!

File: config/routes.rb (hunk 4 above)
Comment (priyapower) [reply config/routes.rb:43]: For sure! Line 43?

Comment (priyapower) [reply config/routes.rb:43]: Done

File: spec/features/cart/show_spec.rb (hunk 5 above)
Comment (priyapower) [reply spec/features/cart/show_spec.rb:7]: I'm glad we learned so much on this user story and set of stories. Shopping cart was intense, but we bonded and learned!!!

**Pull request merged**